### DISCUSSION TITLE
Pull request system

### DISCUSSION THREAD

**Discussion by noahfraiture** [jesseduffield/lazygit#3162] [Q&A]
Hello, I pretty new to open source and I don't exactly how it works. I would like to have the possibility to make merge with squash and I saw that there was already a PR completed with this feature. But I also see that this repo is very popular and there's a lot of issue and PR and the maintener may not have the time. Is there something we can do to accelerate this process or is there another reason why it isn't merge ? 
Maybe the best solution would be to merge it myself and rebuild but I don't know if it's easy to merge an existing PR and I guess the vim extension won't be link and I should also make a fork for it

Thank you for reading me and it's purely curiosity for a newbie :)

**Comment by stefanhaller** [comment]
As for the concrete PR you are talking about: I assume it is #3130? I gave some feedback to the contributor, but didn't hear back yet, so the ball is in their field right now.

In general, you just have to realize that open source work is an unpaid side project for all of us, so we can only put as much time into it as we have, next to our day jobs. Sometimes that's not much, and then things may stall for a while, this is pretty normal for open source work from my experience with other projects. Also, some PRs get reviewed and merged quicker than others, this is also normal, since the complexity of PRs is so different.

As for building from source: that's really easy, you can just make your own local branch where you cherry-pick whatever PRs you want to use. That's what I do, no problem with that. (I'm not sure what vim extension you are talking about and why you think it would have a problem.)

**Reply by noahfraiture** [reply]
Thank you for your answer. 
I of course understand it might not be a priority and it's already amazing to share a project like this. I was indeed talking about #3130 
I'm using a nvim plugin https://github.com/kdheepak/lazygit.nvim but it seems to only be a call to the lazy git on our machine if I understand it well. I will try the local branch system you're talking about. Thank you for your help

**Reply by stefanhaller** [reply]
> but it seems to only be a call to the lazy git on our machine

That shouldn't be a problem. You just have to make sure that the lazygit which you built locally comes before the system one in your PATH. With a standard go setup where you have something like `export PATH=$GOPATH/bin:$PATH` in your shell startup file, this is already the case.

**Comment by noahfraiture** [comment]
How do you cherry pick a PR ? Do you need to see the SHA of all commit in the PR on github and cherry pick them manually or is there a way to see all PR opened in my terminal (in lg ?) and cherry pick them by their # ? 
The pull request are merge from other branch, but I can't directly access to these branch

**Reply by mark2185** [reply]
There are several ways to checkout a pull request, since `lazygit` is on github, the easiest way would probably be to use `gh-cli`, they have [instructions on their website](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/checking-out-pull-requests-locally?tool=cli#modifying-an-active-pull-request-locally).

**Reply by stefanhaller** [reply]
Now this is basic git stuff, really. It seems you need to read a few more tutorials about the basics of git. 😉 

Go to the fork that the contributor has made its PR in; find its git URI; in the case of #3130, that's `git@github.com:rodakd/lazygit.git`. Add that as a remote to your local repo. Now you can see the remote branches of that remote in lazygit, and merge things from there. (Sorry if that was a bit brief, I can't go into more details right now.)

I have to add though: if you really just want to have the command to squash-merge a branch, it's much easier to simply add a custom command for that. It does basically the same thing, and you don't have to mess with local building from source at all. See the comments in #3130 for an example.

**Reply by noahfraiture** [reply]
Ok thank you very much. Indeed I'm pretty much a begineer to git when it's more than add/commit/push
I actually prefer the merge PR method since it can be useful later to know how all of this works
